sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseDecodeData.java

```diff
@@ -59,11 +59,24 @@ default Type getHeadersType() {
      * 2. If the returned int[] is not-null, only the codes in the array are considered as success code.
      *
      * @return the expected HTTP response status codes
+     * @deprecated Use {@link #isExpectedResponseStatusCode(int)} instead.
      */
+    @Deprecated // This is only used in tests now - all uses should go via isExpectedResponseStatusCode instead
     default int[] getExpectedStatusCodes() {
         return null;
     }
 
+    /**
+     * This method returns {@code true} if the given {@code statusCode} is in the list of expected HTTP resposne
+     * codes returned by {@link #getExpectedStatusCodes()}, {@code false} otherwise.
+     *
+     * @param statusCode The HTTP response status code to evaluate.
+     * @return {@code true} if the given status code is expected.
+     */
+    default boolean isExpectedResponseStatusCode(int statusCode) {
```

Comment:
In implementation so I don't think we need a default. If we want to keep a default we should check is the status code is in the 200 range.

`return (statusCode / 100) == 2;`

Reply:
I agree - I thought this was public API so I added it, but I have now removed it again.